src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/jax_rs/ResourceMethodMultipleEntityParamsQuickFix.java

```diff
@@ -47,6 +43,8 @@
 public class ResourceMethodMultipleEntityParamsQuickFix implements IJavaCodeActionParticipant {
 
     private static final Logger LOGGER = Logger.getLogger(ResourceMethodMultipleEntityParamsQuickFix.class.getName());
+    private static final String ENTITY_PARAM_INDEX_KEY = "entityParamIndex";
+    private final List<Integer> entityParamIndexes = new ArrayList<>();
```

Comment:
Each `IJavaCodeActionParticipant` is a singleton. It is not safe to store transient data for a specific code action in this object. Think about what may happen if there are multiple diagnostics that have the same quick fix (e.g. threads that might be simultaneously reading/writing to this object) or what might happen if resolveCodeAction() isn't called immediately after getCodeAction(). Also looks like the field is never cleared, so would leak memory over time. I believe this list should also be stored in the HashMap and carried along with the rest of the `CodeActionResolveData`.

Reply:
Modification done